### PR TITLE
During reparse script context can get closed and destroy debug context, cache script context to check closed status

### PR DESCRIPTION
**Pull request by agarwal-sandeep**
During reparse script context can get closed and destroy debug context, cache script context to check closed status
Only hard fail for reparse failure if scriptcontext is not closed state

### CODE REVIEW

File: lib/Runtime/Base/ScriptContext.cpp [hunk 1]
```diff
@@ -2743,11 +2743,11 @@ if (!sourceList)
 
         hr = this->GetDebugContext()->RundownSourcesAndReparse(shouldPerformSourceRundown, /*shouldReparseFunctions*/ true);
 
-        // Debugger attach/detach failure is catastrophic, take down the process
-        DEBUGGER_ATTACHDETACH_FATAL_ERROR_IF_FAILED(hr);
-
         if (!this->IsClosed())
```
Comment (akroshg) [inline lib/Runtime/Base/ScriptContext.cpp]: I am okay to move the FATAL error inside, but why we are proceeding when the this is closed? why not bail out from this function in that case?

Comment (akroshg) [inline lib/Runtime/Base/ScriptContext.cpp]: Oh - we are not proceeding. I go confused that there is something after this 'if'.

**Comment by akroshg** [comment]
LGTM.

File: lib/Runtime/Base/ScriptContext.cpp (hunk 1 above)
Comment (agarwal-sandeep) [inline lib/Runtime/Base/ScriptContext.cpp]: Everything is inside the !closed check. I will change the check to return hr if closed and move rest of code out of if.

**Pull request merged**